burn-dataset/src/vision/image_folder.rs

```diff
@@ -0,0 +1,382 @@
+use crate::transform::{Mapper, MapperDataset};
+use crate::{Dataset, InMemDataset};
+
+use globwalk::{self, DirEntry};
+use image::{self, ColorType};
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::path::{Path, PathBuf};
+
+const SUPPORTED_FILES: [&str; 4] = ["bmp", "jpg", "jpeg", "png"];
+
+/// Image data type.
+#[derive(Debug, Clone, PartialEq)]
+pub enum PixelDepth {
+    /// 8-bit unsigned.
+    U8(u8),
+    /// 16-bit unsigned.
+    U16(u16),
+    /// 32-bit floating point.
+    F32(f32),
+}
+
+impl TryFrom<PixelDepth> for u8 {
+    type Error = &'static str;
+
+    fn try_from(value: PixelDepth) -> Result<Self, Self::Error> {
+        if let PixelDepth::U8(v) = value {
+            Ok(v)
+        } else {
+            Err("Value is not u8")
+        }
+    }
+}
+
+impl TryFrom<PixelDepth> for u16 {
+    type Error = &'static str;
+
+    fn try_from(value: PixelDepth) -> Result<Self, Self::Error> {
+        if let PixelDepth::U16(v) = value {
+            Ok(v)
+        } else {
+            Err("Value is not u16")
+        }
+    }
+}
+
+impl TryFrom<PixelDepth> for f32 {
+    type Error = &'static str;
+
+    fn try_from(value: PixelDepth) -> Result<Self, Self::Error> {
+        if let PixelDepth::F32(v) = value {
+            Ok(v)
+        } else {
+            Err("Value is not f32")
+        }
+    }
+}
+
+/// Image target for different tasks.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ImageTarget {
+    /// Image-level label.
+    Label(usize),
+    /// Object bounding box.
+    BoundingBox(BoundingBox),
+    /// Segmentation mask.
+    SegmentationMask(SegmentationMask),
+}
+
+/// Segmentation mask target.
+/// For semantic segmentation, a mask has a single channel (C = 1).
+/// For instance segmentation, there may be multiple masks per image (C >= 1).
+#[derive(Debug, Clone, PartialEq)]
+pub struct SegmentationMask {
+    /// Segmentation mask.
+    pub mask: Vec<usize>,
+}
+
+/// Object detection bounding box target.
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub struct BoundingBox {
+    /// Coordinates.
+    pub coords: [f32; 4],
```

Comment:
Are the coordinates `f32` or are they supposed to be int? Not sure what is the most common.